pkg/autoscaler/metrics/http_scrape_client.go

```diff
@@ -45,19 +48,42 @@ func (c *httpScrapeClient) Scrape(url string) (Stat, error) {
 	if err != nil {
 		return emptyStat, err
 	}
+
+	// Ask for protobuf by default. Note that during migration this will not trigger the proto format supported
+	// by Prometheus reporter as the latter uses `application/vnd.google.protobuf`.
+	req.Header.Add("Accept", "application/protobuf")
+
 	resp, err := c.httpClient.Do(req)
 	if err != nil {
 		return emptyStat, err
 	}
+
 	defer resp.Body.Close()
 	if resp.StatusCode < http.StatusOK || resp.StatusCode >= http.StatusMultipleChoices {
 		return emptyStat, fmt.Errorf("GET request for URL %q returned HTTP status %v", url, resp.StatusCode)
 	}
 
-	return extractData(resp.Body)
+	if resp.Header.Get("Content-Type") == "application/protobuf" {
+		return extractDataFromProto(resp.Body)
+	} else {
+		return extractDataFromPrometheusTextFormat(resp.Body)
+	}
+}
+
+func extractDataFromProto(body io.Reader) (Stat, error) {
+	stat := emptyStat
```

Comment:
I'd just use `stat := &Stat{}` here and then call its `Unmarshal` function directly below? :thinking: